modules/cert-manager-acme-http01.adoc

```diff
@@ -0,0 +1,119 @@
+    privateKeySecretRef:
+      name: <secret_for_private_key>                               <2>
+    server: https://acme-staging-v02.api.letsencrypt.org/directory <3>
+    solvers:
+      - http01:
+          ingress:
+            class: openshift-default                               <4>
+----
+<1> Provide a name for the cluster issuer.
+<2> Replace `<secret_private_key>` with the name of secret to store the ACME account private key in.
+<3> Specify the URL to access the ACME server's `directory` endpoint. This example uses the _Let's Encrypt_ staging environment.
+<4> Specify the ingress class. This must be `openshift-default`.
+
+.. Create the `ClusterIssuer` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f acme-cluster-issuer.yaml
+----
+
+. Create an ingress to expose the service of the user workload.
+
+.. Create a YAML file that defines a `Namespace` object:
++
+.Example `namespace.yaml` file
+[source,yaml]
+----
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: my-ingress-namespace <1>
+----
+<1> Specify the namespace for the ingress.
+
+.. Create the `Namespace` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f namespace.yaml
+----
+
+.. Create a YAML file that defines the `Ingress` object:
++
+.Example `ingress.yaml` file
+[source,yaml]
+----
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: sample-ingress                                           <1>
+  namespace: my-ingress-namespace                                <2>
+  annotations:
+    cert-manager.io/cluster-issuer: letsencrypt-staging          <3>
+    acme.cert-manager.io/http01-ingress-class: openshift-default <4>
+spec:
+  ingressClassName: openshift-default                            <5>
+  tls:
+   - hosts:
+       - <ingress_domain_name>                                   <6>
+     secretName: sample-tls                                      <7>
+  rules:
+  - host: <ingress_domain_name>                                  <8>
+    http:
+      paths:
+      - path: /
+        pathType: Prefix
+        backend:
+          service:
+            name: sample-workload                                <9>
+            port:
+              number: 80
+----
+<1> Specify the name of the ingress.
+<2> Specify the namespace that you created for the ingress.
+<3> Specify the cluster issuer that you created.
+<4> Specify the ingress class from the cluster issuer.
+<5> Specify the ingress class from the cluster issuer. (TODO: Update this per PR comment)
+<6> Replace `<ingress_domain_name>` with the ingress domain name. You can retrieve this name by running the following command: `oc get ingresses.config/cluster -o jsonpath={.spec.domain}`.
+<7> Specify the secret to store the created certificate in.
+<8> Replace `<ingress_domain_name>` with the ingress domain name. You can retrieve this name by running the following command: `oc get ingresses.config/cluster -o jsonpath={.spec.domain}`.
```

Comment:
If you mean the value of "host: ingress_domain_name" == ingress domain name == output of "oc get ingresses.config/cluster -o jsonpath={.spec.domain}", then it is wrong. It is the ingress name. You can see https://docs.openshift.com/container-platform/4.12/networking/routes/route-configuration.html#nw-ingress-creating-a-route-via-an-ingress_route-configuration explanation of "host" and my example in https://docs.google.com/document/d/1ZUzGBIZVOwqQrEB-M0zE9aAjK8_ue7Gfsxv_jm4XX7Q/

Reply:
As mentioned in the [comment](https://github.com/openshift/openshift-docs/pull/56754#issuecomment-1469517451) we want the sub-domain prefix (user can provide this) followed by the `cluster_ingress_domain` picked from `oc get dns.config/cluster -o jsonpath={.spec.baseDomain}`.

Reply:
@xingxingxia Does this make it clear?

Reply:
Clear now, great, yeah, good borrowing from from existing doc https://docs.openshift.com/container-platform/4.12/networking/routes/route-configuration.html#nw-ingress-creating-a-route-via-an-ingress_route-configuration about the "host" entry!

Reply:
@xingxingxia @thejasn Are any changes needed with this line, or are we good?

Reply:
@bergerhoffer yes, needed, this line is incorrect as explained in the first message of this thread, you can just copy the sentences from above route doc about the "host" entry shared by Thejas.

Reply:
I did provide a different [suggestion](https://github.com/openshift/openshift-docs/pull/56754#discussion_r1136716509) for `tls.hosts` this can added instead.